src/olympia/addons/models.py

```diff
@@ -1534,6 +1534,12 @@ def blocklistsubmission(self):
         # GUIDs should only exist in one (active) submission at once.
         return BlocklistSubmission.get_submissions_from_guid(self.guid).last()
 
+    @property
+    def git_extraction_is_in_progress(self):
+        if not hasattr(self, 'addongitextraction'):
+            return False
+        return self.addongitextraction.in_progress
```

Comment:
While I do see numerous advantages of using the database for locking, we'll have to make sure that every query checking the "lock" is going to the master to avoid replication lag and ideally it'll play together with the filesystem locking that we already set up in https://github.com/mozilla/addons-server/blob/77e8b8d41e5661029e0c8ebfed535e98f0a98308/src/olympia/files/utils.py#L1354-L1412 even though it's not fully being used by git-extraction yet.

I haven't gone through all the issues you filed for your work though so I may miss some context

Reply:
All the tasks involving this lock will have the `@use_primary_db` decorator. We'll also use the FS lock for the CRON task (to prevent this task to run more than once at the same time).